Csrc/thermodynamics_sa.h

```diff
@@ -61,30 +61,31 @@ void eos_c(struct LookupStruct *LT, double (*lam_fp)(double), double (*L_fp)(dou
             T_2 = T_n;
             f_1 = f_2;
             delta_T  = fabs(T_2 - T_1);
-            
+
             if( isnan(T_n)){
-               printf("%s \n", "Found NaN in T_n"); 
+               printf("%s \n", "Found NaN in T_n");
                printf("%s \t %f\n", "s", s);
                printf("%s \t %f\n", "qt", qt);
-               printf("%s \t %f\n", "p0", p0); 
-               printf("%s \t %f\n", "T_1", T_1); 
-               printf("%s \t %f\n", "T_2", T_2); 
-               printf("%s \t %f\n", "f_1", f_1); 
-               printf("%s \t %f\n", "f_2", f_2); 
-               printf("%s \t %f\n", "L_2", L_2); 
-               printf("%s \t %f\n", "sigma_2", sigma_2); 
-               printf("%s \t %f\n", "lam_2", lam_2); 
-               printf("%s \t %f\n", "s_2", s_2);  
-               printf("%s \t %d\n", "count", count);   
-                
+               printf("%s \t %f\n", "p0", p0);
+               printf("%s \t %f\n", "T_1", T_1);
+               printf("%s \t %f\n", "T_2", T_2);
+               printf("%s \t %f\n", "f_1", f_1);
+               printf("%s \t %f\n", "f_2", f_2);
+               printf("%s \t %f\n", "L_2", L_2);
+               printf("%s \t %f\n", "sigma_2", sigma_2);
+               printf("%s \t %f\n", "lam_2", lam_2);
+               printf("%s \t %f\n", "s_2", s_2);
+               printf("%s \t %d\n", "count", count);
+            }
+            if(count > 1e2){
+               printf("%s \t %d\n", "count", count);
             }
-            
-            
-        } while(delta_T >= 1.0e-3 || sigma_2 < -1.0e-3);
+
+        } while((delta_T >= 1.0e-3 || sigma_2 < -1.0e-3) && count < 1e2);
         *T  = T_2;
         *qv = qv_star_2;
-        *ql = lam_2 * sigma_2;
-        *qi = (1.0 - lam_2) * sigma_2;
+        *ql = lam_fp(T_2) * sigma_2;
+        *qi = (1.0 - lam_fp(T_2)) * sigma_2;
```

Comment:
Do we need to change this two lines? I think `lam_2` is the same as `lam_fp(T_2)`, but I'm fine either way.

Reply:
I think it needs to be updated here